### PR TITLE
Adding printHelloMessage function to HelloWorldPrinter.

### PR DESCRIPTION
**Pull request by KevinWorkman**
Added a `printHelloMessage()` function to the `HelloWorldPrinter` class. This function will be used by the `ExampleMain` class to say hello to users.

### CODE REVIEW

File: src/HelloWorldPrinter.java [hunk 1]
```diff
@@ -10,4 +10,8 @@ public void printHelloWorld() {
     System.out.println("Hello world!");
   }
 
+  public void printHelloMessage(String name) {
+    System.out.println("Hello " + name + "!");
```
Comment (AdaExample) [inline src/HelloWorldPrinter.java]: Should this maybe be `System.err.println` instead?

File: src/HelloWorldPrinter.java [hunk 2]
```diff
@@ -10,4 +10,8 @@ public void printHelloWorld() {
     System.out.println("Hello world!");
   }
 
+  public void printHelloMessage(String name) {
```
Comment (AdaExample) [inline src/HelloWorldPrinter.java]: Can you explain what this code does to me?

Comment (AdaExample) [inline src/HelloWorldPrinter.java]: According to the style guide, public methods should have JavaDoc comments.

Comment (AdaExample) [inline src/HelloWorldPrinter.java]: What happens if `name` is `null`? Should this corner case be handled?

**Review by AdaExample — COMMENTED** [review]
Don't forget to add tests!

File: src/HelloWorldPrinter.java (hunk 2 above)
Comment (KevinWorkman) [reply src/HelloWorldPrinter.java]: Sure, it prints a hello message using the supplied name. For example: `printHelloMessage("Ada")` will print `"Hello Ada!"`.

Comment (KevinWorkman) [reply src/HelloWorldPrinter.java]: Good call. Done.

Comment (KevinWorkman) [reply src/HelloWorldPrinter.java]: In our code this should never happen, but I've added a check for this just in case.

File: src/HelloWorldPrinter.java (hunk 1 above)
Comment (KevinWorkman) [reply src/HelloWorldPrinter.java]: No, `System.err.println()` causes a message to be printed to the standard error output. In our case this will show the message as red text, which we don't want.

**Comment by KevinWorkman** [comment]
I've replied to all comments and added tests. Please take another look as soon as you get a chance.

File: src/HelloWorldPrinter.java [hunk 3]
```diff
@@ -10,4 +10,15 @@ public void printHelloWorld() {
     System.out.println("Hello world!");
   }
 
+  /**
+   * Prints a hello message using the supplied name. For example: printHelloMessage("Ada") will
```
Comment (AdaExample) [inline src/HelloWorldPrinter.java:14]: Add an explanation of what happens when `name` is `null` to the JavaDoc.

File: test/HelloWorldPrinterTest.java [hunk 4]
```diff
@@ -21,4 +21,11 @@ public void testPrintHelloWorld() {
 
     Mockito.verify(mockSystemOut).println("Hello world!");
   }
+
+  @Test
+  public void testPrintHelloMessage() {
```
Comment (AdaExample) [inline test/HelloWorldPrinterTest.java:26]: Add a test for the case when `name` is `null`. That way if anybody takes out the `null` check, the test will fail.

File: src/HelloWorldPrinter.java (hunk 3 above)
Comment (KevinWorkman) [reply src/HelloWorldPrinter.java:14]: Done.

File: test/HelloWorldPrinterTest.java (hunk 4 above)
Comment (KevinWorkman) [reply test/HelloWorldPrinterTest.java]: Done.

Comment (KevinWorkman) [reply test/HelloWorldPrinterTest.java:26]: Done. I wasn't sure how to test that an Exception was thrown, so let me know what you think of my approach.

Comment (AdaExample) [reply test/HelloWorldPrinterTest.java:26]: I wasn't sure either! But your approach makes perfect sense.

**Review by AdaExample — APPROVED** [review]
Thanks for making the changes and explaining your approach to me! LGTM!

**Comment by KevinWorkman** [comment]
Thanks for the review! Merging now.
